app/src/org/commcare/android/models/logic/FormRecordProcessor.java

```diff
@@ -61,12 +63,20 @@ public FormRecord process(FormRecord record)
             FormUploadUtil.getDecryptCipher((new SecretKeySpec(record.getAesKey(), "AES")));
         InputStream is = new CipherInputStream(new FileInputStream(f), decrypter);
 
-        AndroidTransactionParserFactory factory = new AndroidTransactionParserFactory(c, null);
-
-        factory.initCaseParser();
-        factory.initStockParser();
+        AndroidTransactionParserFactory factory = new AndroidTransactionParserFactory(c, null) {
+            @Override
+            public TransactionParser getParser(KXmlParser parser) {
+                String namespace = parser.getNamespace();
+                String name = parser.getName();
+                if (LedgerXmlParsers.STOCK_XML_NAMESPACE.equals(namespace) || "case".equalsIgnoreCase(name)) {
+                    return super.getParser(parser);
+                } else {
+                    return null;
+                }
+            }
+        };
 
-        XmlFormRecordProcessor.process(new AndroidSandbox(CommCareApplication._()), is, factory);
+        XmlFormRecordProcessor.process(null, is, factory);
```

Comment:
The sandbox is an unused arg, so replacing it with `null`. Should fix the commcare code to not require it as an arg (outside of hotfix scope)